File: polocloud-components/component-terminal/src/main/java/dev/httpmarco/polocloud/component/terminal/command/arguments/KeywordArgument.java
package dev.httpmarco.polocloud.component.terminal.command.arguments;

import dev.httpmarco.polocloud.component.terminal.command.CommandArgument;
import dev.httpmarco.polocloud.component.terminal.command.CommandContext;

import java.util.List;

public final class KeywordArgument extends CommandArgument<String> {

    public KeywordArgument(String key) {
        super(key);
    }

    @Override
    public List<String> defaultArgs(CommandContext context) {
        return List.of(key());
    }

    @Override
    public String wrongReason() {
        return "";
    }

    @Override
    public String buildResult(String input) {
        return "";
    }

    @Override
    public boolean predication(String rawInput) {
        return super.predication(rawInput) && rawInput.equalsIgnoreCase(key());
    }
}

File: agent/src/main/kotlin/dev/httpmarco/polocloud/agent/runtime/local/terminal/arguments/type/KeywordArgument.kt
package dev.httpmarco.polocloud.agent.runtime.local.terminal.arguments.type

import dev.httpmarco.polocloud.agent.runtime.local.terminal.arguments.TerminalArgument
import dev.httpmarco.polocloud.agent.runtime.local.terminal.commands.CommandContext

class KeywordArgument(key: String) : TerminalArgument<String>(key) {
    override fun defaultArgs(context: CommandContext): MutableList<String> {
        return listOf(key) as MutableList<String>
    }

    override fun wrongReason(rawInput: String): String {
        return ""
    }

    override fun buildResult(input: String, context: CommandContext): String {
        return ""
    }

    override fun predication(rawInput: String): Boolean {
        return super.predication(rawInput) && rawInput.equals(key, ignoreCase = true)
    }
}